### PR TITLE
Reduce STDERR output during CI

### PR DESCRIPTION
**Pull request by hangy**
**Description:** The default test output is reduced by only logging `error` or above to the TAP. If required for debugging, the `TAP_LOG_FILTER` environment variable can be set to a different level to override it.
**Related issues and discussion:** #3125

### CODE REVIEW

File: t/allergens_tags.t [hunk 1]
```diff
@@ -7,7 +7,6 @@ use utf8;
 
 use Test::More;
 use Log::Any::Adapter 'TAP';
-#use Log::Any::Adapter 'TAP', filter => "none";
```
Comment (stephanegigandet) [inline t/allergens_tags.t]: I'd prefer to keep those commented lines: I keep uncommenting / commenting them when I debug the tests.

File: t/ingredients_nesting.t [hunk 2]
```diff
@@ -309,7 +310,7 @@ foreach my $test_ref (@tests) {
 		# inside the test file much easier when tests results need
 		# to be updated. Caveat is that it might interfere with
 		# test output.
-		or print STDERR join("\n", explain $product_ref->{ingredients});
+		or diag join("\n", explain $product_ref->{ingredients});
```
Comment (stephanegigandet) [inline t/ingredients_nesting.t]: Will we still get this in the test output when the tests fail? We do need it.

Comment (hangy) [reply t/ingredients_nesting.t]: Yes, but better. 😄 `diag` just prints the message, but ensures that the order of messages in the test log is correct.

File: t/allergens_tags.t (hunk 1 above)
Comment (hangy) [reply t/allergens_tags.t]: You can just run `prove` with a different `TAP_LOG_FILTER`, though. Keeping them in the source leaves the risk of accidentally committing them in uncommented form. I actually removed few of these instances in this commit.

**Comment by teolemon** [comment]
fixed a little merge conflict

**Comment by stephanegigandet** [comment]
@hangy I'm not sure why, but all the traces for the script convert_auchan_data.pl now go to STDERR and it makes the test fail.

````
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stderr maxBuffer length exceeded
    at Socket.onChildStderr (child_process.js:386:14)
    at Socket.emit (events.js:311:20)
    at addChunk (_stream_readable.js:294:12)
    at readableAddChunk (_stream_readable.js:271:13)
    at Socket.Readable.push (_stream_readable.js:209:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:186:23) {
  code: 'ERR_CHILD_PROCESS_STDIO_MAXBUFFER',
  cmd: 'perl -c -CS -Ilib scripts/convert_auchan_data.pl',
  stdout: '',

**Comment by hangy** [comment]
> @hangy I'm not sure why, but all the traces for the script convert_auchan_data.pl now go to STDERR and it makes the test fail.
> 
> ```
> RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stderr maxBuffer length exceeded
>     at Socket.onChildStderr (child_process.js:386:14)
>     at Socket.emit (events.js:311:20)
>     at addChunk (_stream_readable.js:294:12)
>     at readableAddChunk (_stream_readable.js:271:13)
>     at Socket.Readable.push (_stream_readable.js:209:10)
>     at Pipe.onStreamRead (internal/stream_base_commons.js:186:23) {
>   code: 'ERR_CHILD_PROCESS_STDIO_MAXBUFFER',
>   cmd: 'perl -c -CS -Ilib scripts/convert_auchan_data.pl',
>   stdout: '',
> ```

The script probably never worked before. `$lc` isn't defined in that line: https://github.com/openfoodfacts/openfoodfacts-server/blob/7aaeb1749fee97ce0e3f6ff80dcfadbb85f0dfa8/scripts/convert_auchan_data.pl#L59
As to why it never showed an error during CI, I do not know.

**Comment by stephanegigandet** [comment]
@hangy : with those changes, will we be testing many more things than before? e.g. all the scripts in /scripts maybe ?

"Failing after 131m "  --> it will be very problematic if tests take 2 hours.

File: t/allergens_tags.t (hunk 1 above)
Comment (stephanegigandet) [reply t/allergens_tags.t:10]: "You can just run prove with a different TAP_LOG_FILTER"
--> how do I do that exactly?

**Comment by hangy** [comment]
> @hangy : with those changes, will we be testing many more things than before? e.g. all the scripts in /scripts maybe ?

No, not really. We're compiling the scripts in `master`, too: https://github.com/openfoodfacts/openfoodfacts-server/blob/46b6657/.github/workflows/pull_request.yml#L84
There might be some hidden issue that makes it appear like everything is okay, but I did intent to ensure that scripts compile, as well.

> "Failing after 131m " --> it will be very problematic if tests take 2 hours.

Agreed. That's much too long. Simply checking the Perl syntax/compiling the scripts really shouldn't take that long, though. So there is probably some hidden issue that causes too much data to be processed during compilation. I'll have a look ...

File: t/allergens_tags.t (hunk 1 above)
Comment (hangy) [reply t/allergens_tags.t:10]: Just set that environment variable when running prove, ie, `TAP_LOG_FILTER=error prove -l -j9`.

**Comment by hangy** [comment]
@stephanegigandet currently, I have no idea what might be causing the build to time out. 🤷‍♂️ I'll look into splitting it up, since some of the changes should make sense on their own.

**Comment by VaiTon** [comment]
@hangy tests failing

**Comment by stephanegigandet** [comment]
t/producers.t                  (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output

-> that's weird, I don't understand why it fails.

**Comment by hangy** [comment]
> -> that's weird, I don't understand why it fails.

Me neither. :) I didn't think I modified the files in that way...

**Comment by VaiTon** [comment]
@hangy do you still want to merge this?

**Pull request closed without merging**